pytest_tests/testsuites/object/test_object_lock.py

```diff
@@ -684,3 +685,85 @@ def test_link_object_of_complex_object_should_also_be_protected_from_deletion(
+                  "(metabase resynchronization must be enabled), and after restarting storage nodes")
+    @pytest.mark.parametrize(
+        "new_locked_storage_object",
+        [pytest.lazy_fixture("simple_object_size"), pytest.lazy_fixture("complex_object_size")],
+        ids=["simple object", "complex object"],
+        indirect=True,
+    )
+    def test_the_object_lock_should_be_kept_after_metabase_deletion(
+            self,
+            new_locked_storage_object: StorageObjectInfo,
+            enable_metabase_resync_on_start,
+    ):
+        """
+        Lock objects should fill metabase on resync_metabase
+        """
+        with allure.step("Log nodes with object"):
+            nodes_with_object = get_nodes_with_object(
+                new_locked_storage_object.cid,
+                new_locked_storage_object.oid,
+                shell=self.shell,
+                nodes=self.cluster.storage_nodes,
+            )
+
+        with allure.step(f"Try to delete object {new_locked_storage_object.oid} before metabase deletion"):
+            with pytest.raises(Exception, match=OBJECT_IS_LOCKED):
+                delete_object(
+                    new_locked_storage_object.wallet_file_path,
+                    new_locked_storage_object.cid,
+                    new_locked_storage_object.oid,
+                    self.shell,
+                    self.cluster.default_rpc_endpoint,
+                )
+
+        with allure.step("Log nodes with object"):
+            nodes_with_object = get_nodes_with_object(
+                new_locked_storage_object.cid,
+                new_locked_storage_object.oid,
+                shell=self.shell,
+                nodes=self.cluster.storage_nodes,
+            )
+
+        with allure.step("Delete metabase files from storage nodes"):
+            for node in self.cluster.storage_nodes:
+                delete_node_metadata(node)
+
+        with allure.step("Start nodes after metabase deletion"):
+            start_storage_nodes(self.cluster.storage_nodes)
+            wait_all_storage_nodes_returned(self.cluster)
+
+        with allure.step("Log nodes with object"):
+            nodes_with_object = get_nodes_with_object(
+                new_locked_storage_object.cid,
+                new_locked_storage_object.oid,
+                shell=self.shell,
+                nodes=self.cluster.storage_nodes,
+            )
+
+        with allure.step(f"Try to delete object {new_locked_storage_object.oid} after metabase deletion"):
+            with pytest.raises(Exception, match=OBJECT_IS_LOCKED):
+                delete_object(
+                    new_locked_storage_object.wallet_file_path,
+                    new_locked_storage_object.cid,
+                    new_locked_storage_object.oid,
+                    self.shell,
+                    self.cluster.default_rpc_endpoint,
+                )
+
+        with allure.step("Restart storage nodes to enable resync_metabase"):
```

Comment:
Is it enabled or disabled here? I thought it's enabled via `enable_metabase_resync_on_start`, so it's active when we `start_storage_nodes(self.cluster.storage_nodes)` above and I'd probably expect it to be disabled then on the node restart (back to normal).

Reply:
> Is it enabled or disabled here? I thought it's enabled via `enable_metabase_resync_on_start`, so it's active when we `start_storage_nodes(self.cluster.storage_nodes)` above and I'd probably expect it to be disabled then on the node restart (back to normal).

Oh, I'm sorry, please. It was a debug, I forgot to delete it.
It's fixed now.